kyuubi-server/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerUserSuite.scala

```diff
@@ -153,4 +154,44 @@ class KyuubiOperationPerUserSuite extends WithKyuubiServer with SparkQueryTests
       }
     }
   }
+
+  test("support to interrupt the thrift request if remote engine is broken") {
+    if (!httpMode) {
+      withSessionConf(Map(
+        KyuubiConf.ENGINE_ALIVE_PROBE_ENABLED.key -> "true",
+        KyuubiConf.ENGINE_ALIVE_PROBE_INTERVAL.key -> "1000",
+        KyuubiConf.ENGINE_ALIVE_TIMEOUT.key -> "3000",
+        KyuubiConf.OPERATION_THRIFT_CLIENT_REQUEST_MAX_ATTEMPTS.key -> "10000"))(Map.empty)(
+        Map.empty) {
+        withSessionHandle { (client, handle) =>
+          val preReq = new TExecuteStatementReq()
+          preReq.setStatement("select engine_name()")
+          preReq.setSessionHandle(handle)
+          preReq.setRunAsync(false)
+          client.ExecuteStatement(preReq)
+
+          val systemExitReq = new TExecuteStatementReq()
+          systemExitReq.setStatement("SELECT java_method('java.lang.Thread', 'sleep', 2000l)," +
+            " java_method('java.lang.System', 'exit', 1)")
+          systemExitReq.setSessionHandle(handle)
+          systemExitReq.setRunAsync(true)
+          client.ExecuteStatement(systemExitReq)
+
+          val executeStmtReq = new TExecuteStatementReq()
+          executeStmtReq.setStatement("SELECT java_method('java.lang.Thread', 'sleep', 30000l)")
+          executeStmtReq.setSessionHandle(handle)
+          executeStmtReq.setRunAsync(false)
+          val startTime = System.currentTimeMillis()
+          val executeStmtResp = client.ExecuteStatement(executeStmtReq)
+          assert(executeStmtResp.getStatus.getStatusCode === TStatusCode.ERROR_STATUS)
+          assert(executeStmtResp.getStatus.getErrorMessage.contains(
+            "java.net.SocketException: Connection reset") ||
+            executeStmtResp.getStatus.getErrorMessage.contains(
+              "Caused by: java.net.SocketException: Broken pipe (Write failed)"))
+          val elapsedTime = System.currentTimeMillis() - startTime
+          assert(elapsedTime < 20 * 1000)
```

Comment:
seems difficult to simulate the socket stuck case.

@yaooqinn  should we remove this UT?

Reply:
create a "socket read timeout" err?

Reply:
ok

Reply:
tried to close the underlying transport socket, but the result is not expected.